app/src/main/kotlin/io/element/android/x/di/SessionComponent.kt

```diff
@@ -33,6 +35,10 @@ interface SessionComponent : NodeFactoriesBindings {
     interface Builder {
         @BindsInstance
         fun client(matrixClient: MatrixClient): Builder
+
+        @BindsInstance
```

Comment:
Why using `@BindsInstance` here? I'd just use `@Provides`

Reply:
Won't `@Provides` need the function to be implemented?

Reply:
We already have lots of @Provides in the `SessionMatrixModule`. I'd use the builder only for things we got from the outer world.

Reply:
Oh, now I see what you mean. Yeah, a provides is better here.